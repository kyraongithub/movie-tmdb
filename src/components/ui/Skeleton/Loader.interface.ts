export interface LoaderPropsInterface {
  radius?: number;
  width: number;
  height: number;
  shape?: 'square' | 'circle';
}
