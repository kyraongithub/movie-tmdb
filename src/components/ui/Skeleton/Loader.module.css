.skeleton {
  border-radius: 0.25rem;
  animation: skeleton-loading 1s linear infinite alternate;
  width: 300px;
  height: 57px;
}

.rounded {
  border-radius: 100%;
}

@keyframes skeleton-loading {
  0% {
    background-color: hsl(200, 20%, 80%);
  }

  100% {
    background-color: hsl(200, 20%, 95%);
  }
}
