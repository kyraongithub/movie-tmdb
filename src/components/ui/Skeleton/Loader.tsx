import classNames from 'classnames';

import styles from './Loader.module.css';
import type { LoaderPropsInterface } from './Loader.interface';

const LoaderSkeleton = (props: LoaderPropsInterface) => {
  const { width, height, shape = 'square', radius = 0 } = props;
  return (
    <div
      className={classNames(styles.skeleton, {
        [styles.rounded]: shape === 'circle',
      })}
      style={{
        width: `${width}px`,
        height: `${height}px`,
        borderRadius: `${radius}px`,
      }}
    />
  );
};

export default LoaderSkeleton;
