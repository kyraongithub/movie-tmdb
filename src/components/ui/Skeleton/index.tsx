import LoaderSkeleton from './Loader';

export default LoaderSkeleton;
