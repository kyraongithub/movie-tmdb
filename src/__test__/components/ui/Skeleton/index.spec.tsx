import { render } from '@testing-library/react';
import Skeleton from '../../../../components/ui/Skeleton';

describe('Skeleton', () => {
  it('render Skeleton component', () => {
    const { container } = render(<Skeleton width={200} height={300} />);
    expect(container).toMatchSnapshot();
  });
});
