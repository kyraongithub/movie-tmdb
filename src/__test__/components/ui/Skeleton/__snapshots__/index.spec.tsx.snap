// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > render Skeleton component 1`] = `
<div>
  <div
    class="_skeleton_8f912d"
    style="width: 200px; height: 300px; border-radius: 0px;"
  />
</div>
`;
